interfaces/acados_matlab_octave/acados_ocp.m

```diff
@@ -120,8 +120,11 @@
                     elseif ~isempty(strfind(obj.opts_struct.qp_solver,'osqp'))
                         flag_file = fullfile(obj.opts_struct.output_dir, '_compiled_with_osqp.txt');
                         compile_interface = ~exist(flag_file, 'file');
+                    elseif ~isempty(strfind(obj.opts_struct.qp_solver,'hpipm'))
```

Comment:
acados is always be compiled with HPIPM, so this is unnecessary.

Reply:
If I understand you correctly, C code compiled with qpOASES will also work if you set HPIPM as your solver. I just wonder if the compiled code would be smaller or more efficient in some way if you build with just HPIPM, i.e. without qpOASES.

Reply:
It is mandatory to compile acados with HPIPM, since HPIPM is used in the core of acados, see e.g.
https://github.com/acados/acados/blob/master/acados/ocp_qp/ocp_qp_common.c#L45
For this reason, there is no need to keep track, if it was compiled with HPIPM.

Then also the `else` branch can set `compile_interface = false;` IMO.
It is basically for the case that, no external QP solver is used, but HPIPM and the interface doesnt have to be recompiled.

The compiled code should be smaller when not compiling with external QP solvers, but not (significantly) more efficient.

Reply:
My changes were mainly focused on this last line in your answer. If you choose HPIPM and it detects existing compilation results with an external QP solver, it will rebuild to remove the parts related to the external solver.
But this is clearly not critical.